python/ray/data/_internal/arrow_block.py

```diff
@@ -238,14 +238,22 @@ def to_numpy(
 
         if columns is None:
             columns = self._table.column_names
-        if not isinstance(columns, list):
+            should_be_single_ndarray = self.is_tensor_wrapper()
+        elif isinstance(columns, list):
+            should_be_single_ndarray = (
+                columns == self._table_column_names and self.is_tensor_wrapper()
+            )
+        else:
             columns = [columns]
+            should_be_single_ndarray = True
```

Comment:
why we don't need to check `columns == self._table_column_names` here?

Reply:
Here, `columns` is a single column string, so we're selecting a single column; this is used internally e.g. in our sort and groupby implementation, and we always want to return a single ndarray in this case. This should never be user-facing, though, all user-facing APIs that eventually use `BlockAccessor.to_numpy()` will either give no argument (for the full table) or a list of columns (for a select-like API).